lib/react/rails/railtie.rb

```diff
@@ -65,7 +65,7 @@ class Railtie < ::Rails::Railtie
       config.after_initialize do |app|
         # Server Rendering
         # Concat component_filenames together for server rendering
-        app.config.react.components_js = lambda {
+        app.config.react.components_js ||= lambda {
```

Comment:
> A naive if would probably suffer from a possible race condition as well.

I din't get you, how over here?

> Does multithreaded Rails not have single-threaded initialization?

No. As a rule of thumb, its better to avoid `||=` in places, where multithreaded execution is expected, something like here. Else it will fail sporadically,